### PR TITLE
Feat/b u toggle report fields

### PR DESCRIPTION
**Pull request by Richard-Pentecost**
#### Work done
- Create checkbox component to select if Nino and Dob can be viewed on downloaded BU report
- Add checkbox to both view last bulk upload and bulk upload main page components
- Add new BulkUploadReport type for a sanitised worker
- Change backend in 3 places to download worker reports with Admin in place of Nino and Dob if checkbox is clicked
- Change upload to retrieve dob and nino from database if uploaded report has Admin instead of values

#### Tests
Does this PR include tests for the changes introduced?
- [x] Yes
- [ ] No, I found it difficult to test
- [ ] No, they are not required for this change

### CODE REVIEW

File: src/app/features/bulk-upload/bulk-upload-page/bulk-upload-page.component.ts [hunk 1]
```diff
@@ -15,21 +16,30 @@ import { AdminSkipService } from '../admin-skip.service';
...
+    private route: ActivatedRoute,
   ) {}
 
-  ngOnInit() {
+  ngOnInit(): void {
     this.breadcrumbService.show(JourneyType.BULK_UPLOAD);
     this.establishment = this.establishmentService.primaryWorkplace;
     this.bulkUploadService.setReturnTo(null);
+    this.isAdmin = this.route.snapshot.data.loggedInUser.role === 'Admin';
+    this.sanitise = this.isAdmin ? true : false;
```
Comment (aaron-russell) [inline src/app/features/bulk-upload/bulk-upload-page/bulk-upload-page.component.ts:35]: Shouldn't this be the permission `canViewNinoDob`?

File: src/app/features/bulk-upload/drag-and-drop-files-list/drag-and-drop-files-list.component.ts [hunk 2]
```diff
@@ -187,17 +190,37 @@ export class DragAndDropFilesListComponent implements OnInit, OnDestroy {
...
 
-  public downloadFile(event: Event, key: string) {
+  public downloadFile(event: Event, key: string): void {
     event.preventDefault();
 
+    // this.bulkUploadService
+    //   .getUploadedFileSignedURL(this.establishmentService.primaryWorkplace.uid, key)
+    //   .subscribe((signedURL) => {
+    //     window.open(signedURL);
+    //   });
```
Comment (aaron-russell) [inline src/app/features/bulk-upload/drag-and-drop-files-list/drag-and-drop-files-list.component.ts:200]: Should this be removed?

File: src/app/features/bulk-upload/drag-and-drop-files-list/drag-and-drop-files-list.component.ts [hunk 3]
```diff
@@ -225,6 +248,7 @@ export class DragAndDropFilesListComponent implements OnInit, OnDestroy {
         }
 
         this.uploadedFiles = uploadedFiles;
+        console.log(this.uploadedFiles);
```
Comment (aaron-russell) [inline src/app/features/bulk-upload/drag-and-drop-files-list/drag-and-drop-files-list.component.ts:251]: Another console log

**Review by aaron-russell — CHANGES_REQUESTED** [review]

File: src/app/features/bulk-upload/drag-and-drop-files-list/drag-and-drop-files-list.component.ts (hunk 3 above)
Comment (Richard-Pentecost) [reply src/app/features/bulk-upload/drag-and-drop-files-list/drag-and-drop-files-list.component.ts:251]: Removed

File: src/app/features/bulk-upload/drag-and-drop-files-list/drag-and-drop-files-list.component.ts (hunk 2 above)
Comment (Richard-Pentecost) [reply src/app/features/bulk-upload/drag-and-drop-files-list/drag-and-drop-files-list.component.ts:200]: removed

File: src/app/core/services/bulk-upload.service.ts [hunk 4]
```diff
@@ -145,6 +145,24 @@ export class BulkUploadService {
     ).pipe(map((response) => response.file.signedUrl));
   }
 
+  public getUploadedFileFromS3(workplaceUid: string, key: string, type: BulkUploadFileType): Observable<any> {
+    const params = new HttpParams().set('downloadType', type);
+
+    console.log('getUploadedFileFromS3 **************');
+    console.log(workplaceUid);
+    console.log(key);
+    console.log(type);
```
[the remark this answers is not in the document]
Comment (Richard-Pentecost) [reply src/app/core/services/bulk-upload.service.ts:154]: Removed

File: src/app/features/bulk-upload/bulk-upload-page/bulk-upload-page.component.ts (hunk 1 above)
Comment (Richard-Pentecost) [reply src/app/features/bulk-upload/bulk-upload-page/bulk-upload-page.component.ts]: Done

**Review by aaron-russell — APPROVED** [review]

**Pull request merged**